adlfs/core.py

```diff
@@ -0,0 +1,424 @@
+    def __init__(self, fs, path, mode='rb', blocksize=2**25, delimiter=None):
+        super().__init__(azure=fs.azure_fs, path=AzureDLPath(path), mode=mode, 
+                         blocksize=blocksize, delimiter=delimiter)
+        self.fs = fs
+        self.path = AzureDLPath(path)
+        self.mode = mode
+    
+    def seek(self, loc, whence=0):
+        """ Set current file location
+
+        Parameters
+        ----------
+        loc: int
+            byte location
+        whence: {0, 1, 2}
+            from start of file, current location or end of file, resp.
+        """
+        loc = int(loc)
+        if not self.mode == "rb":
+            raise ValueError("Seek only available in read mode")
+        if whence == 0:
+            nloc = loc
+        elif whence == 1:
+            nloc = self.loc + loc
+        elif whence == 2:
+            nloc = self.size + loc
+        else:
+            raise ValueError("invalid whence (%s, should be 0, 1 or 2)" % whence)
+        if nloc < 0:
+            raise ValueError("Seek before start of file")
+        self.loc = nloc
+        return self.loc
+
+
+class AzureBlobFileSystem(AbstractFileSystem):
+    """
+    Access Azure Datalake Gen2 as if it were a file system.
+
+    This exposes a filesystem-like API on top of Azure Datalake Gen2 and Azure Storage.
+    
+
+    Examples
+    _________
+    >>> abfs = AzureBlobFileSystem(account_name="XXXX", account_key="XXXX", container_name="XXXX")
+        adl.ls('')
+        
+        Sharded Parquet & csv files can be read as:
+        ----------------------------
+        ddf = dd.read_csv('abfs://container_name/folder/*.csv', storage_options={
+            'account_name': ACCOUNT_NAME, 'account_key': ACCOUNT_KEY,
+        })
+        
+        ddf = dd.read_parquet('abfs://container_name/folder.parquet', storage_options={
+            'account_name': ACCOUNT_NAME, 'account_key': ACCOUNT_KEY,
+        })
+    """
+
+    protocol = 'abfs'
+
+    def __init__(self, account_name: str, container_name: str, account_key: str):
+        
+        """ Access Azure Datalake Gen2 and Azure Storage using Multiprotocol Access
+        
+        Parameters
+        ----------
+        storage_account:  Name of the Azure Storage Account
+        account_key:  Access keys for the Azure Storage account
+        """
+        AbstractFileSystem.__init__(self)
+        self.account_name = account_name
+        self.account_key = account_key
+        self.container_name = container_name
+        self.do_connect()
+    
+    @staticmethod
+    def _get_kwargs_from_urls(paths):
+        """ Get the store_name from the urlpath and pass to storage_options """
+        logging.debug('Getting kwargs from urls...')
+        ops = infer_storage_options(paths)
+        out = {}
+        if ops.get('host', None):
+            out['container_name'] = ops['host']
+        logging.debug(f'kwargs are:  {out}')
+        return out
+    
+    @classmethod
```

Comment:
I wonder how much of this can be placed in a base class that's shared by the two?

Reply:
Is your question about the instantiation or the staticmethods?

Azure Datalake Gen1 doesn't implement access through an access key like is available for Blob storage.

Reply:
Sorry, this was a general comment, not directed at this particular line.